src/components/BookTravelButton.tsx

```diff
@@ -0,0 +1,120 @@
+import {NativeModules} from 'react-native';
+import {useOnyx} from 'react-native-onyx';
+import useLocalize from '@hooks/useLocalize';
+import usePolicy from '@hooks/usePolicy';
+import useThemeStyles from '@hooks/useThemeStyles';
+import {openTravelDotLink} from '@libs/actions/Link';
+import {cleanupTravelProvisioningSession} from '@libs/actions/Travel';
+import Log from '@libs/Log';
+import Navigation from '@libs/Navigation/Navigation';
+import {getAdminsPrivateEmailDomains} from '@libs/PolicyUtils';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
+import Button from './Button';
+import CustomStatusBarAndBackgroundContext from './CustomStatusBarAndBackground/CustomStatusBarAndBackgroundContext';
+import DotIndicatorMessage from './DotIndicatorMessage';
+
+type BookTravelButtonProps = {
+    text: string;
+};
+
+const navigateToAcceptTerms = (domain: string) => {
+    // Remove the previous provision session infromation if any is cached.
+    cleanupTravelProvisioningSession();
+    Navigation.navigate(ROUTES.TRAVEL_TCS.getRoute(domain));
+};
+
+function BookTravelButton({text}: BookTravelButtonProps) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const [activePolicyID] = useOnyx(ONYXKEYS.NVP_ACTIVE_POLICY_ID);
+    const policy = usePolicy(activePolicyID);
+    const [errorMessage, setErrorMessage] = useState('');
+    const [travelSettings] = useOnyx(ONYXKEYS.NVP_TRAVEL_SETTINGS);
+    const [account] = useOnyx(ONYXKEYS.ACCOUNT);
+    const primaryLogin = account?.primaryLogin;
+    const {setRootStatusBarEnabled} = useContext(CustomStatusBarAndBackgroundContext);
+
+    // Flag indicating whether NewDot was launched exclusively for Travel,
+    // e.g., when the user selects "Trips" from the Expensify Classic menu in HybridApp.
+    const [isSingleNewDotEntry] = useOnyx(ONYXKEYS.IS_SINGLE_NEW_DOT_ENTRY);
+
+    const bookATrip = useCallback(() => {
+        setErrorMessage('');
+
+        // The primary login of the user is where Spotnana sends the emails with booking confirmations, itinerary etc. It can't be a phone number.
+        if (!primaryLogin || Str.isSMSLogin(primaryLogin)) {
+            setErrorMessage(translate('travel.phoneError'));
+            return;
+        }
+
+        // An address is required when the Spotnana entity is created for the policy
+        if (isEmptyObject(policy?.address)) {
+            Navigation.navigate(ROUTES.WORKSPACE_PROFILE_ADDRESS.getRoute(policy?.id, Navigation.getActiveRoute()));
+            return;
+        }
+
+        const isPolicyProvisioned = policy?.travelSettings?.spotnanaCompanyID ?? policy?.travelSettings?.associatedTravelDomainAccountID;
+        if (policy?.travelSettings?.hasAcceptedTerms ?? (travelSettings?.hasAcceptedTerms && isPolicyProvisioned)) {
+            openTravelDotLink(policy?.id)
+                ?.then(() => {
```

Comment:
NAB: (and we talked in person about this) but I think it would be better if `openTravelDotLink()` always returned a promise, no matter what, then this optional chaining isn't necessary. That would be easy to implement by doing something like `policy?.id && openTravelDotLink()...` and removing the early return inside of `openTavelDotLink()`